upi/vsphere/cluster_domain/main.tf

```diff
@@ -0,0 +1,36 @@
+locals {
+  lb_hostnames = ["api", "api-int", "*.apps"]
+}
+
+data "aws_route53_zone" "base" {
+  name = var.base_domain
+}
+
+resource "aws_route53_zone" "cluster" {
+  name          = var.cluster_domain
+  force_destroy = true
+
+  tags = {
+    "Name"     = var.cluster_domain
+    "Platform" = "vSphere"
+  }
+}
+
+resource "aws_route53_record" "name_server" {
+  name    = var.cluster_domain
+  type    = "NS"
+  ttl     = "300"
+  zone_id = data.aws_route53_zone.base.zone_id
+  records = aws_route53_zone.cluster.name_servers
+}
+
+resource "aws_route53_record" "a_record" {
+  count = length(local.lb_hostnames)
+
+  type    = "A"
+  ttl     = "60"
+  zone_id = aws_route53_zone.cluster.zone_id
+  name    = "${local.lb_hostnames[count.index]}.${var.cluster_domain}"
+  records = [var.lb_ip_address]
+}
+
```

Comment:
this should be created with the help of host_a_records... because technically lb is running on a machine, it's not different from other machines :)